src/json-patch-duplex.js

```diff
@@ -521,15 +533,9 @@ var jsonpatch;
                         if (validateOperation && operation.op === "add" && key > obj.length) {
                             throw new JsonPatchError("The specified index MUST NOT be greater than the number of elements in the array", "OPERATION_VALUE_OUT_OF_BOUNDS", 0, operation.path, operation);
                         }
-                        if (operation.op === 'test') {
-                            returnValue.test = arrOps[operation.op].call(operation, obj, key, document); // Apply patch
-                            if (returnValue.test === false) {
-                                throw new JsonPatchError("Test operation failed", 'TEST_OPERATION_FAILED', 0, operation, document);
-                            }
-                        }
-                        else {
-                            debugger;
-                            returnValue.removed = arrOps[operation.op].call(operation, obj, key, document); // Apply patch
+                        returnValue = arrOps[operation.op].call(operation, obj, key, document); // Apply patch
```

Comment:
Now, we don't need to create `returnValue` [above](https://github.com/Starcounter-Jack/JSON-Patch/pull/167/commits/0d1d2a9ec9ea9a6559446bf97bf57ce324b3044e#diff-2b195b8e1323c973a44eefb88a63a5c7R447), to replace it here .